frappe/public/js/frappe/form/grid_row.js

```diff
@@ -1122,7 +1122,7 @@ export default class GridRow {
 		if (!field.df.onchange_modified) {
 			var field_on_change_function = field.df.onchange;
 			field.df.onchange = (e) => {
-				field_on_change_function && field_on_change_function(e);
+				field_on_change_function && field_on_change_function(e ?? field);
```

Comment:
Is this frame-work standard operating fallback procedure for onchange if there's no event? (maybe you could link to a code reference) &mdash; if so, this is a no-brainer!

Reply:
Nothing to do, with the code `e.target.value;` works.
Tested in V15.

Reply:
Not sure if I understand more than "it works". :smile: 

To assess the validity of the fix, can you show a quick example:
1. How the `field_on_change_function` would look like when receiving a `e`?
2. How the `field_on_change_function` would look like when receiving a `field`?

Reply:
Apologies for the confusing comment earlier. 

Here's the issue: For "frappe.ui.Dialog", we retrieve the value using "this.value" in the "onchange" method for fields not in the Table link fields. However, for Table link fields, "this.value" doesn't work in the "onchange" method.
After the PR, using the following code: 
`onchange: function(e) { 
	console.log(e.value);  
}`
correctly retrieves the value at the Table link field's "onchange" method.

This issue is specific to Table link fields of "frappe.ui.Dialog." hence Answers to specific scenarios:

1. How the field_on_change_function would look like when receiving a "e"?

Ans:- Undefined.

2. How the field_on_change_function would look like when receiving a "field"?

Ans:- _a5 {df: {…}, parent: j…2.fn.init, only_input: true, with_link_btn: true, doc: {…}, …}

3. How the field_on_change_function would look like when it doesn't know which one it is going to receive?

Ans:- If "e" is available, the system will not receive the "field." Currently, the table link field of "frappe.ui.Dialog" is the only situation where "field" is received, as per the understanding.

Reply:
Thanks a lot, that's really helpful! I appreciate the clarification.

For a design perspective, couldn't we rather make  `frappe.ui.Dialog` comply with the `jquery2.event` api? It could do a try, catch to remain backwards compatible.

Would that be possible? If so, I strongly think it's preferable.